app.rb

```diff
@@ -36,6 +44,16 @@ def to_hash(object)
     hash
   end
 
+  def add_author(_first_name, _last_name)
+    author = Author.new(date, archived)
```

Comment:
I see that you are trying to add an author to the file, this is impressive.  I strongly believe you are trying to add the first name and last name and not date and achieve. I suggest you check it out.
I believe it should be like this: author = `Author.new(first_name, last_name)`